2-database-concurrency-panache/src/main/java/com/manoelcampos/server/rest/ClienteResource.java

```diff
@@ -36,20 +45,20 @@ public long insert(Cliente cliente) {
     @Transactional
     @PUT
     public void update(Cliente cliente) {
-        try{
-            if(Cliente.update(cliente))
-                return;
-        }catch(OptimisticLockException e){
+        try {
+            /**
+             * this is weird, why no CLiente.merge(cliente) ? Missing in the API
+             */
+            Panache.getEntityManager().merge(cliente);
```

Comment:
I think you missed the point.
Getting a direct instance to EntityManager, you're using just plain Hibernate and that sure works. 
`Panache.getEntityManager().merge(cliente)` is not the way it's supposed to work using only the  Panache interfaces. If you check the discussion on the quarkus repo's issue, it's explained why the `merge()` method is not directly exposed on Panache.